apps/full-site-editing/full-site-editing-plugin/donations/edit.js

```diff
@@ -11,34 +11,45 @@ import Donations from './donations';
 import LoadingError from './loading-error';
 import LoadingStatus from './loading-status';
 import UpgradePlan from './upgrade-plan';
+import fetchDefaultProducts from './fetch-default-products';
 import fetchStatus from './fetch-status';
 
 const Edit = () => {
 	const [ isLoading, setIsLoading ] = useState( true );
 	const [ loadingError, setLoadingError ] = useState( '' );
 	const [ shouldUpgrade, setShouldUpgrade ] = useState( false );
 	const [ stripeConnectUrl, setStripeConnectUrl ] = useState( false );
+	const [ products, setProducts ] = useState( [] );
 	const [ upgradeUrl, setUpgradeUrl ] = useState( '' );
 
+	const apiError = ( message ) => {
+		setLoadingError( message );
+		setIsLoading( false );
+	};
+
 	const mapStatusToState = ( result ) => {
 		if ( ( ! result && typeof result !== 'object' ) || result.errors ) {
 			setLoadingError( __( 'Could not load data from WordPress.com.', 'full-site-editing' ) );
 		} else {
 			setShouldUpgrade( result.should_upgrade_to_access_memberships );
 			setUpgradeUrl( result.upgrade_url );
 			setStripeConnectUrl( result.connect_url );
-		}
 
-		setIsLoading( false );
-	};
+			if ( result.products.length ) {
```

Comment:
We'd need to check there are products for the current currency as well. Right now if you remove all the donation-type plans (AFAIK can be done only with `wp shell`) and create a few for a currency other than USD, no default plans for USD will be created.

I also think we need to check that returned products contains all intervals and run `fetchDefaultProducts` if any is missing (the API will create only the missing ones, duplicates prevention is handled in D45327-code).

Reply:
> AFAIK can be done only with wp shell

Ah, found this today: `WP.com API v1.1 POST /sites/:site-id/memberships/product/:product-id/delete` (got it from the Earn UI when deleting from there).

Reply:
Should we add a dev flag for managing this in the Calypso UI? (eg simply return all products in the earn management section) It might be helpful for general debugging

Reply:
f61358962ee8e07ac048ed074d01ec6e3b86f69e should take care of the additional checks. I did notice though, that all three defaults were generated even with existing valid intervals, so I'm not sure if that's a bug with what I've done, or the generation.

Reply:
> that all three defaults were generated even with existing valid intervals

Note that the defaults endpoint will return any valid existing plan. It'll create only missing plans. That means the endpoint will return always the 3 plans, but it doesn't mean all of them were generated during the request.

Reply:
> Note that the defaults endpoint will return any valid existing plan. It'll create only missing plans. That means the endpoint will return always the 3 plans, but it doesn't mean all of them were generated during the request.

I confirmed this by deleting a valid plan using the delete endpoint mentioned above and inserting the block. Only one new plan was created and the endpoint keeps returning the existing ones.